tgui/packages/tgui/interfaces/common/RecipeLookup.js

```diff
@@ -0,0 +1,210 @@
+          <Flex
+            height="50px"
+            width="225px"
+            position="relative"
+            style={{
+              'background-color': 'black',
+            }} >
+            <Chart.Line
+              fillPositionedParent
+              data={recipe.thermodynamics}
+              strokeWidth={0}
+              fillColor={"#3cf072"} />
+            {recipe.explosive && (
+              <Chart.Line
+                position="absolute"
+                top={0.01}
+                bottom={0}
+                right={recipe.isColdRecipe?16.41:0}
+                width="28px"
+                data={recipe.explosive}
+                strokeWidth={0}
+                fillColor={"#d92727"} />
+            )}
+          </Flex>
+        </LabeledList.Item>
+        <LabeledList.Item bold label="Dynamics">
+          <Flex position="relative" top="5px" left="-12px">
+            <Button
+              color="transparent"
+              textColor={recipe.isColdRecipe ? "red" : "white"}
+              content={recipe.isColdRecipe ? recipe.explodeTemp+"K" : recipe.tempMin+"K"}
+              tooltip={recipe.isColdRecipe ? "The temperature at which it is underheated, causing negative effects on the reaction." : "The minimum temperature needed for this reaction to start. Heating it up past this point will increase the reaction rate."}
+              tooltipPosition="right" />
+            {recipe.explosive && (
+              <Flex width="190px" position="relative" top="0px" left="155px" >
+                <Button
+                  color="transparent"
+                  textColor={recipe.isColdRecipe ? "white" : "red"}
+                  content={recipe.isColdRecipe ? recipe.tempMin+"K" : recipe.explodeTemp+"K"}
+                  tooltip={recipe.isColdRecipe ? "The minimum temperature needed for this reaction to start. Heating it up past this point will increase the reaction rate." : "The temperature at which it is overheated, causing negative effects on the reaction."}
+                  tooltipPosition="right" />
+              </Flex>
+            )}
+          </Flex>
+          <LabeledList.Item label="Optimal rate">
+            <Button
+              color="transparent"
+              textColor="white"
+              content={recipe.thermoUpper+"u/s"}
+              tooltip="The fastest rate the reaction can go, in units per second. This is the plateu region shown in the rate profile above."
+              tooltipPosition="right" />
+          </LabeledList.Item>
+          <Flex>
+            <Button
+              color="transparent"
+              textColor="white"
+              content={recipe.thermics}
+              tooltip="The heat generated by a reaction - exothermic produces heat, endothermic consumes heat."
+              tooltipPosition="right" />
+          </Flex>
+        </LabeledList.Item>
+      </LabeledList>
+    ) || (
+      <Box>
+        No reaction selected!
+      </Box>
```

Comment:
Same can be done here, good candidate for early return.

Reply:
same as above